src/app/address/controllers/address/nonUKAddress.test.js

```diff
@@ -0,0 +1,204 @@
+        .returns("Apartment number is required");
+
+      sandbox
+        .stub(address, "getErrorMessage")
+        .callsFake((translate, key, type) => ({
+          text: translate(`${key}.validation.${type}`),
+        }));
+
+      address.getValues(req, res, (err, values) => {
+        expect(err).to.be.null;
+        expect(values).to.deep.include({
+          nonUKAddressApartmentNumberInValid: {
+            text: "Apartment number is required",
+          },
+        });
+        done();
+      });
+    });
+  });
+
+  describe("validateFields", () => {
+    it("defaults validation to the first field when all building address fields are empty", () => {
+      req.body = {
+        nonUKAddressApartmentNumber: "",
+        nonUKAddressBuildingNumber: "",
+        nonUKAddressBuildingName: "",
+      };
+
+      req.form.options.fields = {
+        nonUKAddressApartmentNumber: { validate: [] },
+      };
+
+      sandbox.stub(address, "defaultToFirstField");
+
+      address.validateFields(req, res, next);
+
+      expect(address.defaultToFirstField).to.have.been.calledOnceWith(
+        req.form.options.fields,
+        "nonUKAddressApartmentNumber",
+        req
+      );
+      expect(next).to.have.been.calledOnce;
+    });
+
+    it("does not default to the first field when at least one building address field is provided", () => {
+      req.body = {
+        nonUKAddressApartmentNumber: "2",
+        nonUKAddressBuildingNumber: "",
+        nonUKAddressBuildingName: "",
+      };
+
+      req.form.options.fields = {
+        nonUKAddressApartmentNumber: { validate: [] },
+      };
+
+      sandbox.stub(address, "defaultToFirstField");
+
+      address.validateFields(req, res, next);
+
+      expect(address.defaultToFirstField).not.to.have.been.calledOnceWith(
+        req.form.options.fields,
+        "nonUKAddressApartmentNumber",
+        req
+      );
+      expect(next).to.have.been.calledOnce;
+    });
+  });
+
+  describe("saveValues", () => {
+    it("saves address in the session model", async () => {
+      const addressData = {
+        nonUKAddressApartmentNumber: "Apt 1",
+        nonUKAddressBuildingNumber: "123",
+        nonUKAddressBuildingName: "Example House",
+        nonUKAddressStreetName: "Main Street",
+        nonUKAddressLocality: "Example Town",
+        nonUKAddressPostalCode: "12345",
+        nonUKAddressRegion: "Example Region",
+        nonUKAddressYearFrom: "2020",
+      };
+
+      req.body = addressData;
+      req.sessionModel.get.withArgs("country").returns("FR");
+
+      sandbox.stub(address, "buildAddress").returns({
+        ...addressData,
+        addressCountry: "France",
+      });
+
+      const callback = sandbox.stub();
+      await address.saveValues(req, res, callback);
+
+      expect(req.sessionModel.set).to.have.been.calledWith(
+        "address",
+        sinon.match({
+          ...addressData,
+          addressCountry: "France",
```

Comment:
I'm surpised to see the country display name being saved to the session model here, what was the reason for that?

Reply:
Hmm ok so looking at the [API spec ](https://github.com/govuk-one-login/ipv-cri-address-api/blob/3f3fbb22df0cf69576b4c4c82b23dd5f452fd4ab/infrastructure/lambda/private-api.yaml#L395)`addressCountry` should be the ISO code rather than the full country name